src/Microsoft.Owin.Security.Authorization/AuthorizationDependenciesProvider.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using Microsoft.Owin.Logging;
+using Microsoft.Owin.Security.Authorization.Infrastructure;
+
+namespace Microsoft.Owin.Security.Authorization
+{
+    public class AuthorizationDependenciesProvider
+    {
+        public AuthorizationDependenciesProvider(IAuthorizationPolicyProvider policyProvider = null,
+            IAuthorizationHandler[] handlers = null, ILoggerFactory loggerFactory = null)
+        {
+            OnCreate = options =>
+            {
+                policyProvider = policyProvider ?? new DefaultAuthorizationPolicyProvider(options);
+                handlers = handlers ?? new IAuthorizationHandler[] {new PassThroughAuthorizationHandler()};
+                loggerFactory = loggerFactory ?? new DiagnosticsLoggerFactory();
+                return new AuthorizationDependencies
+                {
+                    PolicyProvider = policyProvider,
+                    Service = new DefaultAuthorizationService(
+                        policyProvider,
+                        handlers,
+                        loggerFactory.Create("ResourceAuthorization"))
+                };
+            };
+            OnDispose = (options, context) => { };
+        }
+
+        public Func<AuthorizationOptions, AuthorizationDependencies> OnCreate { get; set; }
```

Comment:
I dont understand why there are separate injection points (utilizing property dependency injection) for create/dispose pair as opposed to construction DI. What happens if the callers method that sets these properties throws or prematurely exits between setting these two properties, but continues to execute? You now have an object that leaks memory/recourses, and potentially expensive memory/resources.